src/Pages/Joblist/Joblist.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState } from "react";
+import { withRouter } from "react-router";
+import { useParams } from "react-router-dom";
+import CategoryNav from "./Components/CategoryNav/CategoryNav";
+import styled from "styled-components";
+import JoblistMain from "./Components/JoblistMain/JoblistMain";
+import { JOB_API } from "../../config";
+
+const Joblist = location => {
+  const { main, sub } = useParams();
+  const {
+    currentCategories,
+    categoryRoute,
+    reFetchCategory
+  } = useCategoryFetch(main, sub);
+  const { jobData, setJobLoading } = useJobsFetch(main, sub, location);
+
+  return (
+    <Main>
+      <CategoryNav
+        currentCategories={currentCategories}
+        categoryRoute={categoryRoute}
+        reFetch={reFetchCategory}
+        setJobLoading={setJobLoading}
+      />
+      <JoblistMain jobData={jobData} setJobLoading={setJobLoading} />
+    </Main>
+  );
+};
+
+const useCategoryFetch = (main, sub) => {
+  const [categories, setCategories] = useState([]);
+  const [CategoryLoading, setCategoryLoading] = useState(true);
+  const [route, setRoute] = useState([]);
+
+  const filterCategories = allCategories => {
+    let currentCategories = [];
+    if (sub)
+      currentCategories = allCategories[main - 1].sub_category.filter(
+        el => el.id !== +sub
+      );
+    else if (main) currentCategories = allCategories[main - 1].sub_category;
+    else currentCategories = allCategories;
+    setCategories(currentCategories);
+  };
+
+  const getCategoryRoute = allCategories => {
+    let currentCategoryRoute = [];
+    main && currentCategoryRoute.push(allCategories[main - 1]);
+    sub &&
+      currentCategoryRoute.push(
+        allCategories[main - 1].sub_category.filter(el => el.id === +sub)[0]
+      );
+    setRoute(currentCategoryRoute);
+  };
+
+  useEffect(() => {
+    if (CategoryLoading)
+      fetch("http://localhost:3000/data/category_list.json")
```

Comment:
fetch 로컬호스트는 삭제해주세요! 포트번호가 달라질 수 있기 때문에 아래와 같이 작성해주시면 됩니다. 
fetch api는 추후에 변경하시는 거죠?ㅎㅎ

```suggestion
      fetch("/data/category_list.json")
```

Reply:
넹!!